src/modules/data_module.py

```diff
@@ -92,7 +92,39 @@ def __init__(self, source, train_subset_size, test_subset_size,
         # an index 'i' will always refer to the exact same sample
         # this is necessary for easier evaluation later on
         train_all_idx = np.arange(train_size)
-        train_subset_indices = np.random.choice(train_all_idx, size=train_subset_size, replace=False)
+        if balanced_split:
+            classes_idx = {}
+            for target in range(len(self._full_train_set.classes)):
+                idx = np.where(self._full_train_set.targets == target)[0]
+                classes_idx[target] = idx
+
+            sizes = {}
+            if sum(ratio_classes) != 1:
+                raise ValueError("Ratios of the classes should sum to 1")
+            
+            for i, ratio in enumerate(ratio_classes):
+                class_size = int(ratio*train_subset_size)
+                sizes[i] = class_size
+
+            classes_idx_subsets = {}
+            for key in classes_idx:
+                class_indexes = classes_idx[key]
+                
+                if len(class_indexes) == 0:
+                    raise ValueError(f"There are not any indexes connected to class {self._full_train_set.classes[key]}")
+                class_subset_indexes = np.random.choice(class_indexes, size=sizes[key], replace=False)
```

Comment:
```suggestion
                class_subset_indexes = np.random.choice(class_indexes, size=classes_size[key], replace=False)
```
To match previous suggestion